test/spec/controllers/notificationCtrlSpec.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+describe('Notification Controller', function() {
+  var user, scope, rootScope, ctrl;
+
+  beforeEach(function() {
+    user = specHelper.newUser();
+    user._id = "unique-user-id";
+
+    module(function($provide) {
+      $provide.value('User', {user: user});
+      $provide.value('Guide', {});
+    });
+
+    inject(function(_$rootScope_, _$controller_) {
+      scope = _$rootScope_.$new();
+      rootScope = _$rootScope_;
+
+      // Load RootCtrl to ensure shared behaviors are loaded
+      _$controller_('RootCtrl',  {$scope: scope, User: {user: user}});
+
+      ctrl = _$controller_('NotificationCtrl', {$scope: scope, User: {user: user}});
+    });
+  });
+
+  describe('Quest Invitation modal watch', function() {
+    beforeEach(function() {
+      sandbox.stub(rootScope, 'openModal');
+    });
+
+    it('opens quest invitation modal', function() {
+      user.party.quest.RSVPNeeded = true;
+      delete user.party.quest.completed;
+      scope.$digest();
+
+      expect(rootScope.openModal).to.be.calledOnce;
+      expect(rootScope.openModal).to.be.calledWith('questInvitation', {controller:'PartyCtrl'});
+    });
+
+    it('does not open quest invitation modal if RSVPNeeded is not true', function() {
+      user.party.quest.RSVPNeeded = false;
+      delete user.party.quest.completed;
+      scope.$digest();
+
+      expect(rootScope.openModal).to.not.be.called;
+    });
+
+    it('does not open quest invitation modal if quest.completed contains a quest key', function() {
+      user.party.quest.RSVPNeeded = true;
+      user.party.quest.completed = "hedgebeast";
+      scope.$digest();
+
+      // Completion modal should open, but should be the only modal that opens
+      expect(rootScope.openModal).to.be.calledOnce;
+      expect(rootScope.openModal).to.be.calledWith('questCompleted', {controller:'InventoryCtrl'});
```

Comment:
Ah, that's much more intuitive. I'll go ahead and change it, but I just realized that we never explicitly check for the completion modal opening before the invite modal, and this check is testing that behavior. I'll make the suggested change and then move this test to the challenge watch block with a different title.